src/controllers/csi/provisioner/controller.go

```diff
@@ -76,6 +79,27 @@ func (provisioner *OneAgentProvisioner) SetupWithManager(mgr ctrl.Manager) error
 }
 
 func (provisioner *OneAgentProvisioner) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	fs := afero.NewOsFs()
+	if provisioner.maxParallel > 0 {
+		if provisioner.currentParallel == provisioner.maxParallel {
```

Comment:
```suggestion
		if provisioner.currentParallel >= provisioner.maxParallel {
```
a bit more defensive condition :slightly_smiling_face: .